### PR TITLE
PR: Add workflow triggers for release candidates and fix tag normalization

### PR DESCRIPTION
**Pull request by mrclary**
## Description of Changes

* Added workflow trigger for pull requests labeled `build-installer-artifacts`
* Added workflow trigger for manually running installers
* Fix issue where installer version normalized tag version

### Workflow Triggers

Pull requests will continue to trigger all three installer workflows with limited artifacts (Full versions only) and without code-signing or uploading artifacts.

Labeling a pull request with `build-installer-artifacts` or pushing a tag matching the pattern `v*pre*` will also trigger all three installer workflows, but they will run _as if_ it were a release (Lite and Full versions, code-signing, etc.). However, workflow artifacts will not be uploaded to a release. These two trigger methods will not have any affect on a branch in which the latest commit message contains `[ci skip]` (or [equivalent variations](https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs)).

All three installer workflows can be triggered manually from the GitHub web interface or CLI and will execute workflows in the same manner as the aforementioned labeling and tagging mechanisms. This method is not affected by the presence of `[ci skip]` (or variants). However, this workflow trigger mechanism will not be available until this update is propagated to the `master` branch, i.e. after merging to `5.x` and subsequent merge to `master`.

All installer workflows will continue to behave as before upon creating a release.

### Normalized version

`setuptools_scm` by default normalizes our `vX.X.XpreX` release candidate tags to   `vX.X.XrcX`, resulting in an installer named `EXPERIMENTAL-Spyder-5.4.2rc1-macOS-x86_64` instead of `EXPERIMENTAL-Spyder-5.4.2pre1-macOS-x86_64`.

To work around this, we tell `setuptools_scm` not to normalize the version. This results in a version `vX.X.XpreX`, so we further remove the leading `v`, finally arriving at the desired `X.X.XpreX`.

Note that the versions for `external_deps` built locally for the installer must remain normalized; otherwise conda will fail to solve the environment.

### CODE REVIEW

**Comment by ccordoba12** [comment]
Thanks for your work on this @mrclary! We decided with @dalthviz to leave this testing for 5.4.3 to discuss a bit more about some details that are not clear enough for us. Since according to your testing the installers are built just fine for 5.4.2, @dalthviz will proceed with the release now.

**Comment by mrclary** [comment]
@ccordoba12, this should be ready for review now.

File: RELEASE.md
Comment (ccordoba12) [inline RELEASE.md:142]: I don't understand this part: how is the release manager supposed to activate these workflows?

**Comment by ccordoba12** [comment]
> However, this workflow trigger mechanism will not be available until this update is propagated to the master branch, i.e. after merging to 5.x and subsequent merge to master.

You mean after this PR is merged to 5.x and master, right?

Also, it seems you did a lot of testing here, so could you squash some of your commits if that was the case?

**Comment by ccordoba12** [comment]
@dalthviz, I asked your review so that it's clear for you what you need to do in our next release.

File: RELEASE.md
Comment (mrclary) [reply RELEASE.md:142]: The release manager will have to go to the Actions tab in the GitHub repository web interface, select the desired workflow, click the "Run workflow" button, and select the 5.x branch in the popup window. Then repeat for the remaining workflows. See [Running a workflow](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow#running-a-workflow).

Note that the "Run workflow" button is not available until the updated workflow file is merged into `master` branch, so is not currently visible. I've tested this on my fork by merging it into `master` there. Should I include the [Running a workflow](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow#running-a-workflow) link?

File: .github/workflows/installers-conda.yml
Comment (mrclary) [reply .github/workflows/installers-conda.yml:272]: Good question. I'll test it and remove if successful.

File: .github/workflows/installer-win.yml
Comment (mrclary) [reply .github/workflows/installer-win.yml:52]: PRs can have any label, but we want the workflow to behave differently for different pull request actions.

Previously, this workflow triggered on the default `pull_request` types: `opened`, `reopened`, or `sychronize`. This trigger caused the workflow to build developer versions (not "release" versions). In the case of this particular line, this meant that `spyder-kernels` was included in the `UNWANTED_PACKAGES` variable. For "release" versions (i.e. non-PR triggers) `spyder-kernels` would not be included in this variable.

With this PR, we want the behavior of the workflow to remain the same regarding release-like and non-release-like versions, but the trigger now includes the `labeled` `pull_request` type. This changes the logic from `github.event_name == 'pull_request'` to `github.event_name == 'pull_request' && github.event.action != 'labeled'`.

But what about PRs that get labeled an arbitrary label? Well, we don't have to worry about that logic on this line because the `build` job will not run in this case due to line 47. `github.event.action != 'labeled'` on line 47 covers all the previous trigger cases; `github.event.label.name == 'build-installer-artifacts'` on line 47 then permits only the additional `labeled` trigger when the label is correct, i.e. arbitrary labels given to a PR will still trigger this workflow but the jobs will be skipped.

Unfortunately, unlike the `push.tag` trigger, there is no way to filter the workflow trigger on a label pattern, so it must be done at the job level.

File: .github/workflows/installers-conda.yml
Comment (mrclary) [reply .github/workflows/installers-conda.yml]: It is not necessary. I removed this line. Tested in https://github.com/mrclary/spyder/actions/runs/4046651986

Comment (mrclary) [reply .github/workflows/installers-conda.yml]: I merged this job with the build-matrix job to reduce the number of runners. Tested again in https://github.com/mrclary/spyder/actions/runs/4046820255

File: RELEASE.md
Comment (dalthviz) [reply RELEASE.md:142]: I think having some reference info here regarding how to manually trigger the workflows could be nice :+1:

Comment (mrclary) [reply RELEASE.md:142]: I will reference the instructions for manually running a workflow

**Comment by mrclary** [comment]
> > However, this workflow trigger mechanism will not be available until this update is propagated to the master branch, i.e. after merging to 5.x and subsequent merge to master.
> 
> You mean after this PR is merged to 5.x and master, right?

Correct.

> Also, it seems you did a lot of testing here, so could you squash some of your commits if that was the case?

No problem.

**Comment by dalthviz** [comment]
Also, thank you for working on this @mrclary I think this will greatly improve the release process reliability in terms of generated assets/installers :+1:

**Comment by mrclary** [comment]
> Just to be sure I'm understanding correctly, the way the triggering of the workflows will work when doing a release is:
> 
> 
> Right? If I'm understanding this correctly then sounds good to me 👍

That is correct.

> Maybe having some basic set of things/tasks to test manually with the generated installers could be nice to have but I think we can open an issue to add them later to the RELEASE.md

This is a good idea. Presently, the original macOS installer only tests that Spyder.app launches without error. I think we could do the same for the original Windows installer as well as the new conda-based installers.

However, when we move to 6.0(alpha?), we may be able to adapt our existing unit tests to run on a release candidate installation on CI. We can discuss the cost/benefits when we get there...

**Review by dalthviz — APPROVED** [review]

File: .github/workflows/installer-win.yml
Comment (mrclary) [reply .github/workflows/installer-win.yml:52]: @ccordoba12, did this sufficiently answer your question?

Comment (ccordoba12) [reply .github/workflows/installer-win.yml:52]: Yeah, it does. Sorry for the delay

**Review by ccordoba12 — APPROVED** [review]
Thanks @mrclary!

**Pull request merged**
